extensions/sql-database-projects/src/dialogs/deployDatabaseQuickpick.ts

```diff
@@ -0,0 +1,130 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import * as constants from '../common/constants';
+import { AppSettingType, IDeployProfile, ILocalDbSetting } from '../models/deploy/deployProfile';
+import { Project } from '../models/project';
+import * as generator from 'generate-password';
+import { launchPublishDatabaseQuickpick } from './publishDatabaseQuickpick';
+let path = require('path');
+let fse = require('fs-extra');
+
+/**
+ * Create flow for Deploying a database using only VS Code-native APIs such as QuickPick
+ */
+export async function launchDeployDatabaseQuickpick(project: Project): Promise<IDeployProfile | undefined> {
+
+	// Show options to user for deploy to existing server or docker
+	//const deployOption = constants.deployToDockerContainer;
+
+	const deployOption = await vscode.window.showQuickPick(
+		[constants.deployToExistingServer, constants.deployToDockerContainer],
+		{ title: constants.selectDeployOption, ignoreFocusOut: true });
+
+	// Return when user hits escape
+	if (!deployOption) {
+		return undefined;
+	}
+
+	let localDbSetting: ILocalDbSetting | undefined;
+	// Deploy to docker selected
+	if (deployOption === constants.deployToDockerContainer) {
+		let portNumber = await vscode.window.showInputBox(
+			{
+				title: constants.enterPortNumber,
+				ignoreFocusOut: true,
+				value: constants.defaultPortNumber,
+				validateInput: input => isNaN(+input) ? constants.portMustNotBeNumber : undefined
+			}
+		) ?? '';
+
+		// Return when user hits escape
+		if (!portNumber) {
+			return undefined;
+		}
+
+		let password = generator.generate({
+			length: 10,
+			numbers: true,
+			symbols: true,
+			lowercase: true,
+			uppercase: true,
+			exclude: '`' // Exclude the chars that cannot be included in the password. Some chars can make the command fail in the terminal
+		});
+		password = await vscode.window.showInputBox(
+			{
+				title: constants.enterPassword,
+				ignoreFocusOut: true,
+				value: password,
+				password: true
+			}
+		) ?? '';
+
+		// Return when user hits escape
+		if (!password) {
+			return undefined;
+		}
+
+		localDbSetting = {
+			serverName: 'localhost',
+			userName: 'sa',
+			dbName: project.projectFileName,
+			password: password,
+			port: +portNumber,
```

Comment:
what does the + before portNumber do?

Reply:
changing string to number

Reply:
Why don't you just cast it where you get it from the user? Save you having to do that later on like here.